src/Aspire.Hosting.Azure.AppContainers/ContainerAppContext.cs

```diff
@@ -815,13 +816,51 @@ private void AddContainerRegistryParameters(ContainerAppConfiguration app)
 
         app.Registries = [
             new ContainerAppRegistryCredentials
-                    {
-                        Server = _containerRegistryUrlParameter,
-                        Identity = _containerRegistryManagedIdentityIdParameter
-                    }
+            {
+                Server = _containerRegistryUrlParameter,
+                Identity = _containerRegistryManagedIdentityIdParameter
+            }
         ];
     }
 
+    private void AddProbes(ContainerAppContainer containerAppContainer)
+    {
+        if (!resource.TryGetAnnotationsOfType<ProbeAnnotation>(out var probeAnnotations))
+        {
+            return;
+        }
+
+        foreach (var probeAnnotation in probeAnnotations)
+        {
+            ContainerAppProbe? containerAppProbe = null;
+            if (probeAnnotation is EndpointProbeAnnotation endpointProbeAnnotation && endpointProbeAnnotation.EndpointReference.TargetPort.HasValue)
+            {
+                containerAppProbe = new ContainerAppProbe()
+                {
+                    ProbeType = probeAnnotation.Type switch
+                    {
+                        ProbeType.Startup => ContainerAppProbeType.Startup,
+                        ProbeType.Readiness => ContainerAppProbeType.Readiness,
+                        _ => ContainerAppProbeType.Liveness,
+                    },
+                    InitialDelaySeconds = probeAnnotation.InitialDelaySeconds,
+                    PeriodSeconds = probeAnnotation.PeriodSeconds,
+                    HttpGet = new()
+                    {
+                        Path = endpointProbeAnnotation.Path,
+                        Port = endpointProbeAnnotation.EndpointReference.TargetPort.Value,
```

Comment:
We don't want to resolve the target port this way. See https://github.com/dotnet/aspire/blob/f2acbde213f728abb956a1c2adde4120c42be86b/src/Aspire.Hosting.Azure.AppContainers/ContainerAppContext.cs#L496C32-L496C40 for more details on how we do this today for other endpoints.

Reply:
GetValue() works with BicepValue<string> but probe require int. Can I simple parse string into int or there is something with parameter allocation that I'll break?